### PR TITLE
Update to use OCP 4

### PR DESCRIPTION
**Pull request by arilivigni**
- Update dockerfiles
- Add execution of e2e tests

### CODE REVIEW

**Comment by mgencur** [comment]
Just curious. Did you miss to commit some files? The Makefile calls the [e2e tests script](https://github.com/openshift/tektoncd-pipeline/blob/master/Makefile#L14 ) but the script is not in the repository.

File: openshift/e2e-tests-openshift.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+function teardown() {
+  delete_test_namespace
+  delete_test_resources_openshift
+  delete_build_pipeline_openshift
+}
+
+create_test_namespace
+
+enable_docker_schema2
```
Comment (arilivigni) [inline openshift/e2e-tests-openshift.sh]: Do we still need this?

File: openshift/e2e-tests-openshift.sh [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    sed -e 's%github.com/tektoncd/pipeline/cmd/entrypoint%'"$registry_prefix"'\:tektoncd-pipeline-entrypoint%g' | \
+    sed -e 's%github.com/tektoncd/pipeline/cmd/git-init%'"$registry_prefix"'\:tektoncd-pipeline-git-init%g' | \
+    sed -e 's%github.com/tektoncd/pipeline/cmd/kubeconfigwriter%'"$registry_prefix"'\:tektoncd-pipeline-kubeconfigwriter%g' | \
+    sed -e 's%github.com/tektoncd/pipeline/cmd/nop%'"$registry_prefix"'\:tektoncd-pipeline-nop%g' \
+    >> $resolved_file_name
+    echo >> $resolved_file_name
+  done
+}
+
+function enable_docker_schema2(){
```
Comment (arilivigni) [inline openshift/e2e-tests-openshift.sh]: Do we still need this?

File: openshift/e2e-tests-openshift.sh (hunk 1 above)
Comment (vdemeester) [reply openshift/e2e-tests-openshift.sh]: Looking in `knative-serving`, we don't :angel:

**Comment by vdemeester** [comment]
Alright, https://github.com/openshift/release/pull/3341 should enable `tide` in there. As this is green, I'll manually merge this PR and work on releases branch and master updates

**Pull request merged**